### DISCUSSION TITLE
What is the default Fetcher function?

### DISCUSSION THREAD

**Discussion by thiagonunesbatista** [vercel/swr#910] [General] [answered]
I started reading the documentation of SWR and I discovered that exists an optional param to useSwr called fetcher (a function to fetch data). If I just pass the key to useSwr and don't pass a fetcher,  what function or code will be executed to fetch the data?

```
const { data, error } = useSwr('https://api.chucknorris.io/jokes/random')
```

**Comment by promer94** [comment, marked as answer]
https://swr.vercel.app/docs/options#options
```js
fetcher = window.fetch(url).then(res => res.json())
```

**Comment by kraynel** [comment]
This is no longner the case since SWR 1.0. The default fetcher function is now `null`, which trigger no network call whatsoever.
https://swr.vercel.app/blog/swr-v1#no-more-default-fetcher
